server/watch_volfile_changes.py

```diff
@@ -0,0 +1,127 @@
+import os
+import time
+import json
+import logging
+import serverutils
+from kadalulib import (CommandException, execute, logf,
+                       logging_setup)
+from glusterfsd import (create_brick_volfile, create_client_volfile)
+from shd import create_shd_volfile
+
+VOLINFO_DIR = "/var/lib/gluster"
+VOLFILES_DIR = "/var/lib/kadalu/volfiles"
+
+brick_device = os.environ.get("BRICK_DEVICE", None)
+brick_path = os.environ["BRICK_PATH"]
+
+volume_id = os.environ["VOLUME_ID"]
+brick_path_name = brick_path.strip("/").replace("/", "-")
+volname = os.environ["VOLUME"]
+nodename = os.environ["HOSTNAME"]
+
+volfile_id = "%s.%s.%s" % (volname, nodename, brick_path_name)
+
+storage_unit_volfile_path = os.path.join(VOLFILES_DIR, "%s.vol" % volfile_id)
+client_volfile_path = os.path.join(VOLFILES_DIR, "%s.vol" % volname)
+shd_volfile_path = os.path.join(VOLFILES_DIR, "glustershd.vol")
+
+data = {}
+info_file_path = os.path.join(VOLINFO_DIR, "%s.info" % volname)
+
+def send_sighup():
+    """ Send SIGHUP to Brick & SHD Process notifing change in volfiles """
```

Comment:
May be calling shell command easier?

```
out, err = execute("pgrep -f  -a 'glusterfsd|shd')
for line in out.split("\n"):
    pid = line.split(" ", 1)[0]
    os.kill(int(pid), signal.SIGHUP)
```

Log if any errors.

Reply:
Updated